// @fluentui/react-toolbar/lib-commonjs/components/ToolbarButton/useToolbarButtonStyles.styles.raw.js
'use client';
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "useToolbarButtonStyles_unstable", {
    enumerable: true,
    get: function() {
        return useToolbarButtonStyles_unstable;
    }
});
const _react = require("@griffel/react");
const _reactbutton = require("@fluentui/react-button");
const useBaseStyles = (0, _react.makeStyles)({
    vertical: {
        flexDirection: 'column'
    },
    verticalIcon: {
        fontSize: '24px',
        margin: '0'
    }
});
const useToolbarButtonStyles_unstable = (state)=>{
    'use no memo';
    const buttonStyles = useBaseStyles();
    // eslint-disable-next-line react-hooks/immutability
    state.root.className = (0, _react.mergeClasses)(state.vertical && buttonStyles.vertical, state.root.className);
    if (state.icon) {
        // eslint-disable-next-line react-hooks/immutability
        state.icon.className = (0, _react.mergeClasses)(state.vertical && buttonStyles.verticalIcon, state.icon.className);
    }
    (0, _reactbutton.useButtonStyles_unstable)(state);
};
